linera-views/src/key_value_store_view.rs

```diff
@@ -83,12 +86,20 @@ where
     }
 
     async fn load(context: C) -> Result<Self, ViewError> {
-        let key = context.base_tag(KeyTag::Hash as u8);
-        let hash = context.read_key(&key).await?;
+        let key1 = context.base_tag(KeyTag::Hash as u8);
+        let key2 = context.base_tag(KeyTag::Sizes as u8);
+        let keys = vec![key1, key2];
+        let values_bytes = context.read_multi_key_bytes(keys).await?;
+        let hash = from_bytes_opt(&values_bytes[0])?;
+        let key_sizes = match &values_bytes[1] {
+            None => BTreeMap::new(),
+            Some(bytes) => bcs::from_bytes(bytes)?,
```

Comment:
Do we need to store/load all the keys at once?

Incidentally, BCS (de)serialization of maps and sets is actually rather expensive because canonical ordering of keys is enforced. It should be faster to deserialize as `Vec<(Vec<u8>, usize)>` (same bytes as a map) then convert to map.

Reply:
Yes, we need to store/load all the keys at once.

This is because we need to be able to access the keys when computing the effect of the batch (or other operations) on the complete set of keys. We need that for:
* Accessing the `stored_size`.
* Writing the batch with the `flush`. 
The problem with the second operation is that `flush` is not async so it cannot access the data.

Maybe all of this could be avoided, but it does not seem simple right now.

Thank you for the BCS serialization. I will do it accordingly.